staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go

```diff
@@ -35,6 +36,12 @@ func ErrBadArgType(expected, actual string) error {
 	return fmt.Errorf("expected a %s, but received a %s", expected, actual)
 }
 
+func ErrBadArgTypeAssertion(expected, actual interface{}) error {
```

Comment:
Looks like the one with Nil can't be replaced easily, but maybe the error isn't super correct either anyway.